kimchi/src/circuits/polynomials/foreign_mul/foreign_mul_0.rs

```diff
@@ -0,0 +1,91 @@
+/// ForeignMul0 - Foreign field element constraints
+///
+///    Foreign field element F is comprised of three 88-bit limbs L0L1L2
+///
+///    * This circuit gate is used to partially constrain L0 or L1
+///    * The rest of L0 and L1 are constrained by a single ForeignMul2
+///    * It operates on the Curr row
+///
+/// Column | Curr
+///      0 | L
+///      1 | plookup Lp0
+///      2 | plookup Lp1
+///      3 | plookup Lp2
+///      4 | plookup Lp3
+///      5 | copy Lp4
+///      6 | copy Lp5
+///      7 | crumb Lc0
+///      8 | crumb Lc1
+///      9 | crumb Lc2
+///     10 | crumb Lc3
+///     11 | crumb Lc4
+///     12 | crumb Lc5
+///     13 | crumb Lc6
+///     14 | crumb Lc7
+use std::marker::PhantomData;
+
+use crate::circuits::{
+    argument::{Argument, ArgumentType},
+    expr::{witness_curr, E},
+    gate::GateType,
+    polynomial::COLUMNS,
+};
+use ark_ff::{FftField, Zero};
+
+use super::common::{sublimb_crumb_constraint, two};
+
+#[derive(Default)]
+pub struct ForeignMul0<F>(PhantomData<F>);
+
+impl<F> Argument<F> for ForeignMul0<F>
+where
+    F: FftField,
+{
+    const ARGUMENT_TYPE: ArgumentType = ArgumentType::Gate(GateType::ForeignMul0);
+    const CONSTRAINTS: u32 = 9;
+
+    // Constraints for ForeignMul0
+    //   * Operates on Curr row
+    //   * Range constrain all sublimbs except p4 and p5 (baring plookup constraints, which are done elsewhere)
+    //   * Constrain that combining all sublimbs equals the limb stored in column 0
+    fn constraints() -> Vec<E<F>> {
+        // Row structure
+        //  Column w(i) 0    1       ... 4       5    6    7     ... 14
```

Comment:
`Column w(i)` and `Curr` don't feel like they're adding much here apart from visual noise. Delete?